engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingformula/BaseRollingFormulaOperator.java

```diff
@@ -0,0 +1,291 @@
+        final WritableChunk<? extends Values> outputValues;
+        final BiConsumer<Object, Integer> outputSetter;
+
+        @SuppressWarnings("unused")
+        protected Context(final int affectedChunkSize, final int influencerChunkSize) {
+            outputFillContext = outputSource.makeFillFromContext(affectedChunkSize);
+            outputValues = outputChunkType.makeWritableChunk(affectedChunkSize);
+            outputSetter = getChunkSetter(outputValues);
+        }
+
+        @Override
+        public void accumulateCumulative(
+                @NotNull final RowSequence inputKeys,
+                @NotNull final Chunk<? extends Values>[] valueChunkArr,
+                @Nullable final LongChunk<? extends Values> tsChunk,
+                final int len) {
+            throw new UnsupportedOperationException("RollingFormula is not supported in cumulative operations.");
+        }
+
+        void writeNullToOutputChunk(final int outIdx) {
+            outputSetter.accept(nullValue, outIdx);
+        }
+
+        @Override
+        public void writeToOutputColumn(@NotNull final RowSequence inputKeys) {
+            outputSource.fillFromChunk(outputFillContext, outputValues, inputKeys);
+        }
+
+        @Override
+        public void reset() {
+            nullCount = 0;
+        }
+
+        @Override
+        public void close() {
+            outputValues.close();
+            outputFillContext.close();
+        }
+    }
+
+    public BaseRollingFormulaOperator(
+            @NotNull final MatchPair pair,
+            @NotNull final String[] affectingColumns,
+            @Nullable final String timestampColumnName,
+            final long reverseWindowScaleUnits,
+            final long forwardWindowScaleUnits,
+            @NotNull final String formula,
+            @NotNull final String paramToken,
+            @NotNull final Map<Class<?>, FormulaColumn> formulaColumnMap,
+            @NotNull final TableDefinition tableDef) {
+        super(pair, affectingColumns, timestampColumnName, reverseWindowScaleUnits, forwardWindowScaleUnits, true);
+        this.formulaColumnMap = formulaColumnMap;
+        this.tableDef = tableDef;
+
+        final String outputColumnName = pair.leftColumn;
+
+        final Class<?> columnType = tableDef.getColumn(pair.rightColumn).getDataType();
+        vectorType = getVectorType(columnType);
+
+        // Handle the rare (and probably not useful) case where the formula is an identity formula. We need to make
+        // a copy of the RingBuffer wrapper and store that as a DirectVector. If not, we will point to the live data
+        // in the ring.
+        final String formulaToUse = formula.equals(paramToken) ? formula + ".getDirect()" : formula;
+
+        // Re-use the formula column if it's already been created for this type. No need to synchronize; these
+        // operators are created serially.
+        // TODO: does generic Object need to be handled uniquely?
```

Comment:
I can't think of a reason why an `Object` based formula column couldn't be memoized. The user will need to handle type checking on their own, and each `Object` column will need the same formula to be invoked anyway.